firstvoices/backend/views/contact_us_views.py

```diff
@@ -0,0 +1,174 @@
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    create=extend_schema(
+        description="Sends emails to the addresses listed on the site contact_us_email and contact_us_users fields.",
+        responses={
+            202: OpenApiResponse(description="Success. Email successfully sent."),
+            403: OpenApiResponse(description=doc_strings.error_403_site_access_denied),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+        request=inline_serializer(
+            name="Contact Us Form",
+            fields={
+                "name": serializers.CharField(),
+                "email": serializers.CharField(),
+                "message": serializers.CharField(),
+            },
+        ),
+    ),
+)
+class ContactUsView(
+    SiteContentViewSetMixin,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    GenericViewSet,
+):
+    http_method_names = ["get", "post"]
+    serializer_class = ContactUsSerializer
+    pagination_class = None
+
+    def get_queryset(self):
+        return self.get_validated_site()
+
+    def list(self, request, *args, **kwargs):
+        """
+        Returns the list of receiver emails used by the contact us post endpoint.
+        """
+
+        site = self.get_validated_site().first()
+        perm = Site.get_perm("change")
+        if not request.user.has_perm(perm, site):
+            return Response({"message": error_403}, status=403)
+        return super().list(request, *args, **kwargs)
+
+    def create(self, request, *args, **kwargs):
+        """
+        Sends emails to the addresses listed on the site contact_us_email and contact_us_users fields. A list of
+        fallback email addresses can be set in the AppJson model with the key 'contact_us_default_emails'. If the
+        site does not have any contact emails or users, the fallback email will be used. Additionally, the site can
+        specify a list of excluded words in the AppJson model with the key 'contact_us_excluded_words'. If an email
+        contains any of these words in the email, name, or message it will not be sent.
+        """
+
+        logger = logging.getLogger(__name__)
+        site = self.get_validated_site().first()
+        if request.user.is_anonymous:
+            return Response(
+                {"message": "You must be logged in to send an email."}, status=403
+            )
+        try:
+            to_email_list = site.contact_email
+            to_email_list.extend(site.contact_users.values_list("email", flat=True))
+            name = request.data["name"]
+            from_email = request.data["email"]
+            message = request.data["message"]
+
+            # Validate that the email field from the form is a valid email address
+            validate_email(from_email)
```

Comment:
Will this fail if the email address is blank or missing? Can you please add a test to verify that this field is required?